--- syslog-ng-3.12.1/modules/java/proxies/java-logmsg-proxy.c.orig	2017-09-20 21:02:23.000000000 +0200
+++ syslog-ng-3.12.1/modules/java/proxies/java-logmsg-proxy.c	2017-10-24 20:13:28.068958005 +0200
@@ -25,6 +25,7 @@
 #include "java-logmsg-proxy.h"
 #include "java_machine.h"
 #include "messages.h"
+#include "str-utils.h"
 #include "logmsg/logmsg.h"
 
 
@@ -49,6 +50,7 @@
 {
   LogMessage *msg = (LogMessage *)handle;
   const gchar *value;
+  gssize value_len;
 
   const char *name_str = (*env)->GetStringUTFChars(env, name, NULL);
   if (name_str == NULL)
@@ -56,12 +58,13 @@
       return NULL;
     }
 
-  value = log_msg_get_value_by_name(msg, name_str, NULL);
+  value = log_msg_get_value_by_name(msg, name_str, &value_len);
 
   (*env)->ReleaseStringUTFChars(env, name, name_str);
 
   if (value)
     {
+      APPEND_ZERO(value, value, value_len);
       return (*env)->NewStringUTF(env, value);
     }
   else
